spec/rubocop/cop/rspec/repeated_example_group_body_spec.rb

```diff
@@ -0,0 +1,216 @@
+  end
+
+  it 'does not register offense for different block body implementation' do
+    expect_no_offenses(<<-RUBY)
+      context 'when awesome case' do
+        it { cool_predicate_method }
+      end
+
+      context 'when another awesome case' do
+        it { another_predicate_method }
+      end
+    RUBY
+  end
+
+  it 'does not register offense if metadata is different' do
+    expect_no_offenses(<<-RUBY)
+      describe 'doing x' do
+        it { cool_predicate_method }
+      end
+
+      describe 'doing x', :request do
+        it { cool_predicate_method }
+      end
+    RUBY
+  end
+
+  it 'does not register offense with several docstring' do
+    expect_no_offenses(<<-RUBY)
+      describe 'doing x', :json, 'request' do
+        it { cool_predicate_method }
+      end
+
+      describe 'doing x', 'request' do
+        it { cool_predicate_method }
+      end
+    RUBY
+  end
+
+  it 'registers offense for different groups' do
+    expect_offense(<<-RUBY)
+      describe 'doing x' do
+      ^^^^^^^^^^^^^^^^^^^^^ Repeated describe block body on line(s) [5]
+        it { cool_predicate_method }
+      end
+
+      context 'when a is true' do
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated context block body on line(s) [1]
+        it { cool_predicate_method }
+      end
+    RUBY
+  end
+
+  it 'does not register offense for example groups in different groups' do
+    expect_no_offenses(<<-RUBY)
+      describe 'A' do
+        describe '.b' do
+          context 'when this' do
+            it { do_something }
+          end
+        end
+        context 'when this' do
+          it { do_something }
+        end
+      end
+    RUBY
+  end
+
+  it 'registers offense only for RSPEC namespace example groups' do
+    expect_offense(<<-RUBY)
+      helpers.describe 'doing x' do
+        it { cool_predicate_method }
+      end
+
+      RSpec.describe 'doing x' do
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated describe block body on line(s) [9]
+        it { cool_predicate_method }
+      end
+
+      context 'when a is true' do
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated context block body on line(s) [5]
+        it { cool_predicate_method }
+      end
+    RUBY
+  end
+
+  it 'registers offense only for RSPEC namespace example groups in any order' do
+    expect_offense(<<-RUBY)
+      RSpec.describe 'doing x' do
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated describe block body on line(s) [5]
+        it { cool_predicate_method }
+      end
+
+      context 'when a is true' do
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Repeated context block body on line(s) [1]
+        it { cool_predicate_method }
+      end
+
+      helpers.describe 'doing x' do
+        it { cool_predicate_method }
+      end
+    RUBY
+  end
+
+  it 'registers offense only for example group' do
+    expect_offense(<<-RUBY)
+      RSpec.describe 'A' do
+        stub_all_http_calls()
+        allowed_statuses = %i[open submitted approved].freeze
+        before { create(:admin) }
```

Comment:
Since those are not duplicated, it is not clear how this test proves its description